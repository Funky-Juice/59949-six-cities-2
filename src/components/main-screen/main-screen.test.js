import {BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import offerTestObj from '../../mocks/test-offer';
import MainScreen from './index';

const mockOffers = [offerTestObj];
const createNodeMock = () => document.createElement(`div`);

const store = createStore(() => ({
  offers: mockOffers,
  activeOffers: mockOffers,
  activeCity: mockOffers[0].city
}));

it(`MainScreen correctly renders`, () => {
  const options = {createNodeMock};
  const tree = renderer
    .create(<Provider store={store}>
      <Router>
        <MainScreen/>
      </Router>
    </Provider>, options)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
