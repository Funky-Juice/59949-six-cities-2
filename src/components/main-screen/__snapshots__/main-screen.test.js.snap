// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen correctly renders 1`] = `
<main
  className="page__main page__main--index "
>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="#"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container container "
    >
      <section
        className="cities__places places"
      >
        <b
          className="places__found"
        >
          1
           
          place
           to stay in 
          Amsterdam
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by
          </span>
          <span
            className="places__sorting-type"
            onClick={[Function]}
            style={
              Object {
                "paddingLeft": "5px",
              }
            }
          >
            Popular
            <svg
              className="places__sorting-arrow"
              height="4"
              style={
                Object {
                  "transform": "rotateX(0deg)",
                }
              }
              width="7"
            >
              <use
                xlinkHref="#icon-arrow-select"
              />
            </svg>
          </span>
          <ul
            className="places__options places__options--custom places__options--opened"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <li
              className="places__option places__option--active"
              onClick={[Function]}
            >
              Popular
            </li>
            <li
              className="places__option false"
              onClick={[Function]}
            >
              Price: low to high
            </li>
            <li
              className="places__option false"
              onClick={[Function]}
            >
              Price: high to low
            </li>
            <li
              className="places__option false"
              onClick={[Function]}
            >
              Top rated first
            </li>
          </ul>
        </form>
        <div
          className="cities__places-list places__list tabs__content"
        >
          <article
            className="cities__place-card place-card"
            onMouseEnter={[Function]}
          >
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="room.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    80
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="
          button
          place-card__bookmark-button
          place-card__bookmark-button--active
        "
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  >
                    3
                  </span>
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/2"
                >
                  Wood and stone place
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                room
              </p>
            </div>
          </article>
        </div>
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
        >
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </section>
      </div>
    </div>
  </div>
</main>
`;
